### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by masci**
Preview: 
- [contributing.md](https://github.com/deepset-ai/haystack-core-integrations/blob/massi/contrib/CONTRIBUTING.md)
- [README](https://github.com/deepset-ai/haystack-core-integrations/blob/massi/contrib/README.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+> in order to pass.
+
+If you want to run only a portion of the tests, for example including integration tests, Hatch will happily take the
+same options you would pass to `pytest` directly, in this case a marker with the option `-m`:
+
+```console
+$ hatch run test -m"not integration"
+```
+
+### Improving The Documentation
```
Comment (masci) [inline CONTRIBUTING.md:238]: @dfokina can you have a look at this section?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+There are two types of documentation for this project:
+
+1. Python API docs, and
+2. Documentation pages
+
+The Python API docs detail the source code: classes, functions and parameters that implement every integration.
```
Comment (dfokina) [inline CONTRIBUTING.md:245]:
```suggestion
The Python API docs detail the source code: classes, functions, and parameters that every integration implements.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+This type of documentation is extracted from the source code itself, and contributors should pay attention when they
+change the code to also change relevant comments and docstrings. This type of documentation is mostly useful to
+developers, but it can be handy for users at times. You can browse it on the dedicated section in the
+[documentation website](https://docs.haystack.deepset.ai/v2.0/reference/integrations-chroma).
+
+Documentation pages explain what an integration accomplishes and explain how to use it and how to configure it. This
```
Comment (dfokina) [inline CONTRIBUTING.md:251]:
```suggestion
Documentation pages explain what an integration accomplishes, how to use it, and how to configure it. This
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+type of documentation mostly targets end-users, and contributors are welcome to make changes and keep it up-to-date.
+You can contribute changes to the documentation pages using the "Suggest Edits" button that each page shows on its
+top-right corner.
```
Comment (dfokina) [inline CONTRIBUTING.md:254]:
```suggestion
You can contribute changes to the documentation using the "Suggest Edits" button in the top-right corner of each page.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+## Suggesting Enhancements
+
+This section guides you through submitting an enhancement suggestion, including new integrations and improvements
+to existing ones. Following these guidelines will help maintainers and the community to understand your suggestion and find related suggestions.
+
+
+### Before Submitting an Enhancement
+
+- Make sure that you are using the latest version.
+- Read the [documentation]() carefully and find out if the functionality is already covered, maybe by an individual configuration.
```
Comment (dfokina) [inline CONTRIBUTING.md:106]: Link is missing :)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+- Perform a [search](/issues) to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider writing an and distributing the integration on your own.
```
Comment (dfokina) [inline CONTRIBUTING.md:108]:
```suggestion
- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider writing and distributing the integration on your own.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+
+### Improving The Documentation
+
+There are two types of documentation for this project:
+
+1. Python API docs, and
+2. Documentation pages
+
+The Python API docs detail the source code: classes, functions and parameters that implement every integration.
+This type of documentation is extracted from the source code itself, and contributors should pay attention when they
```
Comment (dfokina) [inline CONTRIBUTING.md:246]: Do we also want to explain that pydoc config files are needed to display the docs on the website when a new integration is added?

**Review by dfokina — COMMENTED** [review]
Thanks for including the docs, Massi! I left some language suggestions to make it clearer, but also wondering if we want to explain working with the pydoc config files?

File: CONTRIBUTING.md (hunk 7 above)
Comment (masci) [reply CONTRIBUTING.md:246]: Yes let's do it

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+
+### Before Submitting a Bug Report
+
+A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to
+investigate carefully, collect information and describe the issue in detail in your report. Please complete the
+following steps in advance to help us fix any potential bug as fast as possible.
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://docs.haystack.deepset.ai/v2.0/docs/intro). If you are looking for support, you might want to check [this section](#i-have-a-question)).
+- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](issues).
```
Comment (silvanocerza) [inline CONTRIBUTING.md:71]:
```suggestion
- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](/issues).
```

**Pull request merged**